### DISCUSSION TITLE
Single devtools for tanstack query/router/etc... ?

### DISCUSSION THREAD

**Discussion by yann-combarnous** [TanStack/query#7737] [Ideas] [answered]
Right now, when using Tanstack router and Tanstack query, devtools don't really play well together. 

What would be nice is a common devtools icon and a tab-based container for each Tanstack module: router, query, etc...

**Comment by dvtkrlbs** [comment]
I would also very much prefer a browser extension based tanstack extension that can bundle all of them

**Comment by andreeasteriu** [comment, marked as answer]
You can now use TanStackDevtools from @tanstack/react-devtools and configure it in the following way:
```Node
<TanStackDevtools
      config={{
        position: 'bottom-right',
      }}
      plugins={[
        {
          name: 'Tanstack Router',
          render: <TanStackRouterDevtoolsPanel />,
        },
        {
          name: 'React Query',
          render: <ReactQueryDevtoolsPanel />,
        },
        formDevtoolsPlugin(),
      ]}
    />
```
